poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -31,13 +31,37 @@ pub struct Heartbeats {
 
 impl Heartbeats {
     pub async fn validated(
-        exec: impl sqlx::PgExecutor<'_> + Copy,
+        exec: impl sqlx::PgExecutor<'_>,
         starting: DateTime<Utc>,
     ) -> std::result::Result<Self, sqlx::Error> {
         let heartbeats =
             sqlx::query_as::<_, Heartbeat>("SELECT * FROM heartbeats WHERE timestamp >= $1")
-                .bind(starting.naive_utc())
-                .fetch(exec);
+                .bind(starting)
+                .fetch_all(exec)
+                .await?
+                .into_iter()
+                .map(
+                    |Heartbeat {
+                         hotspot_key,
```

Comment:
This is a handful to read. Would a `impl (HeartbeatKey, HeartbeatValue) from  Heartbeat` work?

Reply:
Unfortunately that chunk has to go somewhere (since it's a heartbeat row), but I think I improved it slightly by not destructuring the heartbeat